File: v2/src/main/java/com/api/v2/customers/domain/CustomerAuditTrail.java
package com.api.v2.customers.domain;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;
import java.time.ZoneId;
import java.util.UUID;

@Document(collection = "customer_audit_trail")
public record CustomerAuditTrail(
        @Id
        UUID id,
        Customer customer,
        LocalDateTime createdAt,
        ZoneId createdAtZone
) {

    public static CustomerAuditTrail of(Customer customer) {
        return new CustomerAuditTrail(
                UUID.randomUUID(),
                customer,
                LocalDateTime.now(),
                ZoneId.systemDefault()
        );
    }

}

File: v2/src/main/kotlin/com/api/v2/customers/CustomerAuditTrail.kt
package com.api.v2.customers

import java.time.LocalDateTime
import java.time.ZoneId
import java.util.UUID

data class CustomerAuditTrail(
    val id: UUID,
    val customer: Customer,
    val createdAt: LocalDateTime,
    val createdAtZone: ZoneId
)
